lib/Command/PreGenerate.php

```diff
@@ -200,19 +200,18 @@ private function processFile(File $file) {
 			}
 
 			try {
-				foreach ($this->sizes['square'] as $size) {
-					$this->previewGenerator->getPreview($file, $size, $size, true);
-				}
-
-				// Height previews
-				foreach ($this->sizes['height'] as $height) {
-					$this->previewGenerator->getPreview($file, -1, $height, false);
-				}
-
-				// Width previews
-				foreach ($this->sizes['width'] as $width) {
-					$this->previewGenerator->getPreview($file, $width, -1, false);
-				}
+				$specifications = array_merge(
+					array_map(function ($squareSize) {
+						return ['width' => $squareSize, 'height' => $squareSize, 'crop' => true];
+					}, $this->sizes['square']),
+					array_map(function ($heightSize) {
+						return ['width' => -1, 'height' => $heightSize, 'crop' => false];
+					}, $this->sizes['height']),
+					array_map(function ($widthSize) {
+						return ['width' => $widthSize, 'height' => -1, 'crop' => false];
+					}, $this->sizes['width'])
+				);
```

Comment:
I know this is very old, but `$specifications` doesn't depend on the current file, so it could be computed only once and set as class member